DNN Platform/Library/Services/Zip/SharpZipLibRedirect.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+
+namespace DotNetNuke.Services.Zip
+{
+    /// <summary>
+    /// This class contains a fix for a breaking upgrade in DNN 9.2 regarding the ZIP.
+    /// It is in charge of fixing an assembly rename which was necessary when DNN
+    /// upgraded from an older SharpZipLib with a different DLL name. 
+    /// </summary>
+    /// <remarks>
+    /// The class in this folder will be used by.net when an assembly is missing.
+    /// It will then check if .net was looking for SharpZipLib, and if necessary,
+    /// redirect it to the correct (new) assembly.
+    ///
+    /// Once this remapping has been completed, this code will not be used again until the next
+    /// restart of the DNN application. 
+    /// 
+    /// final solution taken from https://raw.githubusercontent.com/2sic/2sxc/master/2sxc%20Dnn/Dnn920/SharpZipLibRedirect.cs
+    /// </remarks>
+    internal class SharpZipLibRedirect
+    {
+        internal static bool AlreadyRun { get; private set; }
```

Comment:
Hm, I notice this changed from `public` to `internal`. But does it really have to be accessible from anywhere? I'd expect such property to be `private`

Reply:
I would leave it in for debugging - in case this is ever questioned. But I can change it if that is preferred.

Reply:
No strong opinion here. Let's leave it as is